React week2

elicarcar opened the pull request.

ArcoMul commented on homework-submission/react-week2/src/App.js:
```diff
@@ -0,0 +1,72 @@
+import TodoItems from './components/TodoItems'
+import "./App.css";
+
+
```
Too much whitespace 😬

ArcoMul commented on homework-submission/react-week2/src/App.js:
```diff
@@ -0,0 +1,72 @@
+    setTodoItems(newList)
+  }
+
+  return <div className="App">
```
Your jsx formatting is a bit strange. Do you have prettier installed?

ArcoMul commented on homework-submission/react-week2/src/App.js:
```diff
@@ -0,0 +1,72 @@
+    {
+      name: "dont fall over"
+    }
+  ]);
```
It would be more neat to keep track if a todo item is done or not here in the todo (sub)-objects. So that you have all the data in one place. The code of marking a todo item done would be easier to read that way.

ArcoMul commented on homework-submission/react-week2/src/components/HyfButton.js:
```diff
@@ -0,0 +1,9 @@
+    return (
+       
+            <button className="add-btn" onClick={onHandleClick}>Add</button>
+        
```
Mind your white space

ArcoMul commented on homework-submission/react-week2/src/App.js:
```diff
@@ -0,0 +1,72 @@
+
+  return <div className="App">
+  <TodoList 
+  input={<HyfInput onHandleInputChange={onHandleInputChange}/>} 
```
Why are you passing components to the props of `TodoList`? Here you can pass just the function `onHandleInputChange` and let the `TodoList` render it's own input field. For `children` counts the same, no need to already pass rendered components, better to pass the array of values and let the child component render it's own children.

ArcoMul commented on homework-submission/react-week2/src/components/TodoItems/index.js:
```diff
@@ -0,0 +1,18 @@
+
+
+    return (
+        <li style={isDone()}>
```
Better to give a className based on if the item is done or not. This way you have a function which seems to be general (`isDone`), but actually returns a very specific styling value.

elicarcar commented:
Hi @ArcoMul thank you for the feedback! And im so sorry, i guess it must be so annoying to read someone else's bad-structured code! You are right, last week i had a problem with my code formatter, i fixed that issue so now all the white spaces and unnecessary gaps should be gone. I changed my checkbox and modified 'IsDone' as well. I just couldn't change my nested components (the issue you're indicating between lines 57- 59 ).

ArcoMul commented:
Super, great improvements! :clap:
